src/modules/AutoinstScripts.rb

```diff
@@ -305,11 +308,46 @@ def Import(s)
       s = deep_copy(s)
       Builtins.y2debug("Calling AutoinstScripts::Import()")
       # take only hash entries (bnc#986049)
-      @pre = s.fetch("pre-scripts", []).select { |h| h.is_a?(Hash) }
-      @init = s.fetch("init-scripts", []).select { |h| h.is_a?(Hash) }
-      @post = s.fetch("post-scripts", []).select { |h| h.is_a?(Hash) }
-      @chroot = s.fetch("chroot-scripts", []).select { |h| h.is_a?(Hash) }
-      @postpart = s.fetch("postpartitioning-scripts", []).select { |h| h.is_a?(Hash) }
+      @pre = s.fetch("pre-scripts", []).select do |h|
+          if h.is_a?(Hash)
+            true
+          else
+            log.warn "Cannot evaluate pre-script: #{h}"
+            false
+          end
+        end
+      @init = s.fetch("init-scripts", []).select do |h|
+          if h.is_a?(Hash)
+            true
+          else
+            log.warn "Cannot evaluate init-script: #{h}"
+            false
+          end
+        end
+      @post = s.fetch("post-scripts", []).select do |h|
+          if h.is_a?(Hash)
+            true
+          else
+            log.warn "Cannot evaluate post-script: #{h}"
+            false
+          end
+        end
+      @chroot = s.fetch("chroot-scripts", []).select do |h|
+          if h.is_a?(Hash)
+            true
+          else
+            log.warn "Cannot evaluate chroot-script: #{h}"
+            false
+          end
+        end
+      @postpart = s.fetch("postpartitioning-scripts", []).select do |h|
+          if h.is_a?(Hash)
+            true
+          else
+            log.warn "Cannot evaluate postpartitioning-script: #{h}"
+            false
+          end
+        end
```

Comment:
I see a lot of duplication here. I really suggest to have own method for it. like
```
def valid_scripts_for(tree, key)
  tree.fetch(key, []).select do |h|
    next true if h.is_a?(Hash)

   log.warn "Cannot evaluate #{key}: #{h.inspect}"
   false
end

# usage
@postpart = valid_scripts_for(s, "postpartitioning-scripts")
```